src/functions/common.php

```diff
@@ -370,3 +373,22 @@ function drupal_get_library(string $module, ?string $name = null): array|false
 
     return $libraryDiscovery->getLibrariesByExtension($module);
 }
+
+
+/**
+ * @param array<string, mixed> $options
+ */
+function drupal_goto(string $path = '', array $options = [], int $http_response_code = 302): void
+{
+    \Drupal::moduleHandler()->alter('drupal_goto', $path, $options, $http_response_code);
+    $url = \Drupal::pathValidator()->getUrlIfValidWithoutAccessCheck($path);
+    if ($url !== false) {
+        $url->mergeOptions($options);
+        $goto = $url->toString();
+        if ($goto instanceof GeneratedUrl) {
+            $goto = $goto->getGeneratedUrl();
+        }
+        $response = new RedirectResponse($goto, $http_response_code);
+        throw new EnforcedResponseException($response);
+    }
```

Comment:
@darrenoh does this approach work as expected for you? It's leveraging what the Form API does to avoid the verbosity of 

```
        $request = \Drupal::request();
        $request->getSession()->save();
        $response->prepare($request);
        \Drupal::service('kernel')->terminate($request, $response);
        $response->send();
```

Reply:
Yes, that works.